src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -437,12 +439,14 @@ exprt string_constraint_generatort::add_axioms_for_parse_int(
     not_exprt(or_exprt(starts_with_minus, starts_with_plus));
 
   /// TODO: we should throw an exception when this does not hold:
-  exprt correct=add_axioms_for_correct_number_format(str, radix);
+  std::size_t max_string_length=40;
```

Comment:
I always assumed that the `--string-max-length` option for CBMC was CBMC saying "I won't of my own accord make strings longer than this". Is this true, or will it truncate string literals as well? Because putting `--string-max-length` here could easily break `Long.parseLong(print(long)) == long`.